client/flutter/lib/pages/main_pages/app_tab_router.dart

```diff
@@ -49,20 +60,65 @@ class AppTabRouter extends StatelessWidget {
     );
   }
 
+  void _switchPage(int index) {
+    if (currentIndex != index) {
+      setState(() {
+        currentIndex = index;
+      });
+    }
+  }
+
+  bool _onSwitchTabNotification(SwitchTabNotification notification) {
+    final String routeName = notification.route;
+    final int newIndex = widget.tabs.indexWhere(
+      (Widget Function(BuildContext) builder) =>
```

Comment:
This is very clever, but I think it's too clever, and could be fragile in the future.

Reply:
I initially wanted to provide a `Type` to `SwitchTabNotification` but the dynamic nature of the home page content makes it hard. 

I'd be willing to improve it e.g. by whitelisting route names of pages that should exhibit this behavior.

It's also an easy one to remove if it goes wrong, either stop dispatching notifications or always use the navigator when notifications are intercepted.